### DISCUSSION TITLE
Are passwords created with auth.signin already encrypted?

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#437] [Questions] [answered]
I started to use bcrypt to send passwords, but I realized I didn't know how to compare passwords or if you even could do such a thing.

Do I not need to hash passwords?

**Comment by inian** [comment, marked as answer]
Yes, the [passwords are hashed](https://github.com/supabase/gotrue/blob/d536ed5ae341e818d8e1d83e83d237ad3c556e4a/models/user.go#L198) with bcrypt before they are stored in the database. You needn't do that in the application layer.
